app/setup_handlers.go

```diff
@@ -30,34 +30,6 @@ func setUpgradeHandler(app *ElysApp) {
 			app.Logger().Info("Running upgrade handler for " + version.Version)
 
 			if version.Version == NextVersion || version.Version == LocalNetVersion {
-
-				// delete all mtps
-				mtps := app.PerpetualKeeper.GetAllLegacyMTPs(ctx)
-				for _, mtp := range mtps {
-					app.PerpetualKeeper.DestroyMTP(ctx, sdk.MustAccAddressFromBech32(mtp.Address), mtp.Id)
-				}
-
-				// delete all perpetual pools
-				perpPools := app.PerpetualKeeper.GetAllLegacyPools(ctx)
-				for _, pool := range perpPools {
-					app.PerpetualKeeper.RemovePool(ctx, pool.AmmPoolId)
-				}
-
-				// delete all accounted pools
-				accountedPools := app.AccountedPoolKeeper.GetAllAccountedPool(ctx)
-				for _, pool := range accountedPools {
-					app.AccountedPoolKeeper.RemoveAccountedPool(ctx, pool.PoolId)
-				}
-
-				// initiate accounted pools
-				pools := app.AmmKeeper.GetAllPool(ctx)
-				for _, pool := range pools {
-					err := app.AccountedPoolKeeper.InitiateAccountedPool(ctx, pool)
-					if err != nil {
-						panic(err)
-					}
-				}
-
```

Comment:
This was added recently(not included in upgrade), are we sure we don't need this ?

Reply:
I am doing it through migrations. No need to InitiateAccountedPool here or in migration, it gets trigger in hooks in enable pool tx of perpetual